crates/typst-ide/src/jump.rs

```diff
@@ -113,25 +113,30 @@ pub fn jump_from_cursor(
     document: &Document,
     source: &Source,
     cursor: usize,
-) -> Option<Position> {
+) -> Vec<Position> {
     fn is_text(node: &LinkedNode) -> bool {
         node.get().kind() == SyntaxKind::Text
     }
 
     let root = LinkedNode::new(source.root());
-    let node = root
+    let Some(node) = root
         .leaf_at(cursor, Side::Before)
         .filter(is_text)
-        .or_else(|| root.leaf_at(cursor, Side::After).filter(is_text))?;
+        .or_else(|| root.leaf_at(cursor, Side::After).filter(is_text))
+    else {
+        return vec![];
+    };
 
     let span = node.span();
-    for (i, page) in document.pages.iter().enumerate() {
-        if let Some(point) = find_in_frame(&page.frame, span) {
-            return Some(Position { page: NonZeroUsize::new(i + 1).unwrap(), point });
-        }
-    }
-
-    None
+    document
+        .pages
+        .iter()
+        .enumerate()
+        .filter_map(|(i, page)| {
+            find_in_frame(&page.frame, span)
```

Comment:
I think if we're already doing this, we probably want to find multiple hits in the same frame as well. So `find_in_frame` could take an `&mut Vec<Position>` as its first argument or something like that (would need to pass the page number down for this).

Reply:
Though, hmm, we would probably get a bunch of hits closeby for glyphs coming from the same span. Not sure how to handle that. Text might be chunked up into `FrameItem::Text` somewhat arbitrarily...

Reply:
I'll let you decide because I have no intuition for these things @laurmaedje

Reply:
I think one per page is fine because we have no current use case for more.